### PR TITLE
feat: support to struct datatype

### PR DESCRIPTION
**Pull request by WenyXu**
Support to struct datatype

#14

### CODE REVIEW

File: src/arrow_reader/column.rs [hunk 1]
```diff
@@ -19,19 +20,24 @@ pub mod float;
 pub mod int;
 pub mod present;
 pub mod string;
+pub mod r#struct;
```
Comment (Jefffrey) [inline src/arrow_reader/column.rs:23]: Could we rename to not have to use raw identifiers? Maybe struct_column for example

Comment (WenyXu) [reply src/arrow_reader/column.rs:23]: Yes, I will rename it.

File: src/arrow_reader.rs [hunk 2]
```diff
@@ -143,6 +152,499 @@ pub enum Decoder {
...
+
+    Ok(())
+}
+
+impl Decoder {
+    pub fn new_array_builder(&self, capacity: usize) -> Box<dyn ArrayBuilder> {
+        match self {
+            Decoder::Int64(_) => Box::new(PrimitiveBuilder::<Int64Type>::with_capacity(capacity)),
+            Decoder::Int32(_) => Box::new(PrimitiveBuilder::<Int64Type>::with_capacity(capacity)),
+            Decoder::Int16(_) => Box::new(PrimitiveBuilder::<Int64Type>::with_capacity(capacity)),
```
Comment (Jefffrey) [inline src/arrow_reader.rs:318]: Are these types correct?

**Review by Jefffrey — APPROVED** [review]
Looks good 👍

Lots of changes but we can probably focus on getting functionality in first before then optimizing/refactoring later

File: src/arrow_reader.rs (hunk 2 above)
Comment (WenyXu) [reply src/arrow_reader.rs:318]: Let me double-check this part.

**Pull request merged**